### PR TITLE
e2e: make tests pod's image configurable

### PR DESCRIPTION
**Pull request by shajmakh**
Running the tests on disconnected environments might lead to pod crash while pulling images from external registries.

Keep using Pause image as the default test image and allow configuring the image via an environment variable TEST_IMAGE.

### CODE REVIEW

File: test/e2e/serial/workload_placement_test.go [hunk 1]
```diff
@@ -158,7 +158,7 @@ var _ = Describe("[serial][disruptive][scheduler] workload placement", func() {
 				Expect(err).ToNot(HaveOccurred(), "cannot detect the zone ID from %q", zone.Name)
 				// now put a minimal pod (1 cpu 1Gi) on both zones. Now the target node as whole will still have the
 				// required resources, but no NUMA zone individually will
-				targetedPaddingPod := objects.NewTestPodPause(fxt.Namespace.Name, fmt.Sprintf("tgtpadpod-%s", zone.Name))
+				targetedPaddingPod := objects.NewTestPod(fxt.Namespace.Name, fmt.Sprintf("tgtpadpod-%s", zone.Name))
```
Comment (fromanirh) [inline test/e2e/serial/workload_placement_test.go:161]: renaming this function is unnecessary - and I'd said wrong in this context

File: test/utils/images/images.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+package images
+
+import (
+	"os"
+
+	"github.com/openshift-kni/numaresources-operator/test/utils/objects"
+)
+
+var testImage string
```
Comment (fromanirh) [inline test/utils/images/images.go:21]: besides my general dislike for globals and `init` (unless where is clearly needed, and here is not), we can simply have something like
```
func GetPauseImage() string {
       if pullSpec, ok := os.LookupEnv("E2E_PAUSE_IMAGE_URL"); ok {
               return pullSpec
       }
       return objects.PauseImage
}
```
and remove everything else

File: test/utils/objects/pod.go [hunk 3]
```diff
@@ -37,9 +38,16 @@ func NewTestPodPause(namespace, name string) *corev1.Pod {
 			TerminationGracePeriodSeconds: &zero,
 			Containers: []corev1.Container{
 				{
-					Name:    name + "-cnt",
-					Image:   PauseImage,
-					Command: []string{PauseCommand},
+					Name:  name + "-cnt",
+					Image: images.Test(),
+					Command: []string{
```
Comment (fromanirh) [inline test/utils/objects/pod.go:43]: this change breaks the compatibility with the `pause` image, which is something I'd avoid just yet. Let's just revert this change. The function should read like
```
func NewTestPodPause(namespace, name string) *corev1.Pod {
	var zero int64
	return &corev1.Pod{
		ObjectMeta: metav1.ObjectMeta{
			Name:      name,
			Namespace: namespace,
		},
		Spec: corev1.PodSpec{
			TerminationGracePeriodSeconds: &zero,
			Containers: []corev1.Container{
				{
					Name:    name + "-cnt",
					Image:   images.GetPauseImage(),
					Command: []string{PauseCommand},
				},
			},
		},
	}
}
```

File: test/utils/objects/pod.go [hunk 4]
```diff
@@ -20,13 +20,14 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/openshift-kni/numaresources-operator/test/utils/images"
```
Comment (fromanirh) [inline test/utils/objects/pod.go:23]: the conventional import grouping is
```
stdlib

3rd parties (most known, usually kube)

3rd parties (less known, usually github packages)

local
```
so please move this import in its own group at the bottom of the `import` block

**Review by fromanirh — COMMENTED** [review]
thanks for the PR!

File: test/e2e/serial/workload_placement_test.go (hunk 1 above)
Comment (shajmakh) [reply test/e2e/serial/workload_placement_test.go:161]: thanks for your comments.
why do we want to limit the use only for Pause image in the name of the function?

File: test/utils/objects/pod.go (hunk 4 above)
Comment (shajmakh) [reply test/utils/objects/pod.go]: thank you, I'll note this for next time. updated

File: test/e2e/serial/workload_placement_test.go (hunk 1 above)
Comment (fromanirh) [reply test/e2e/serial/workload_placement_test.go:161]: Because I think is not time yet to generalize the function. We still want a container which does `pause` and sits idle

Comment (shajmakh) [reply test/e2e/serial/workload_placement_test.go]: ok, I see. updated. thanks!

File: test/utils/images/images.go [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package images
+
+import "os"
```
Comment (fromanirh) [inline test/utils/images/images.go:19]: ok, so you moved the `PauseImage` const in the `test/utils/images` package. I don't feel strongly about that. But then you need to either revert the change or to import the package here...

File: test/utils/images/images.go [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+func GetPauseImage() string {
+	if pullSpec, ok := os.LookupEnv("E2E_PAUSE_IMAGE_URL"); ok {
+		return pullSpec
+	}
+	return PauseImage
```
Comment (fromanirh) [inline test/utils/images/images.go:25]: ...and fix the name here and in all the relevant places as reported by go vet/lint.
Up to you, I'm fine both ways.

File: test/utils/images/images.go (hunk 5 above)
Comment (shajmakh) [reply test/utils/images/images.go:19]: importing `test/utils/objects` caused cyclic import which isn't allowed in Go.

**Review by fromanirh — COMMENTED** [review]
/lgtm
/approve
on a followup PR, let's make sure we can use the -tests image (either independently or integrated into something else) as pause image
